### DISCUSSION TITLE
About TemporalDataLoader in TGN

### DISCUSSION THREAD

**Discussion by Everyday-seu** [pyg-team/pytorch_geometric#4499] [General]
Is len(data) correct? In the example you gave in tgn:
train_data, val_data, test_data = data.train_val_test_split(val_ratio=0.15, test_ratio=0.15)
train_loader = TemporalDataLoader(train_data, batch_size=200)
But len(train_data)=5,so len(data)=5 and it turns to the "arange" becomes(0,5,batch_size). Is it correct to use something like data.num_events?
Hope for your reply! Thanks!

**Comment by rusty1s** [comment]
Len(data) refers to the number of events in TemporalData, see https://github.com/pyg-team/pytorch_geometric/blob/master/torch_geometric/data/temporal.py#L148. We felt that overriding this property for temporal data made a lot of sense.

**Reply by Everyday-seu** [reply]
Thanks for your reply!
I know why I have this problem now. I find that my temporal.py is not the same with your temporal.py. Is it because you update the code while I use the old ones?

**Reply by rusty1s** [reply]
Yes, it looks like that. Updating to the latest PyG version should fix this :)

**Reply by Everyday-seu** [reply]
Thanks! It works :)
